pkg/container-utils/containerd/containerd.go

```
@@ -15,31 +15,246 @@
+	if err != nil {
+		return nil, fmt.Errorf("listing containers: %w", err)
+	}
+
+	ret := make([]*runtimeclient.ContainerData, 0, len(containers))
+	for _, container := range containers {
+		if isSandboxContainer(container) {
+			log.Debugf("ContainerdClient: container %q is a sandbox container. Temporary skipping it", container.ID())
+			continue
+		}
+
+		task, err := getContainerdTask(container)
+		if err != nil {
+			return nil, err
+		}
+
+		containerData, err := taskAndContainerToContainerData(task, container)
+		if err != nil {
+			return nil, err
+		}
+
+		ret = append(ret, containerData)
+	}
+
+	return ret, nil
+}
+
+func (c *ContainerdClient) GetContainer(containerID string) (*runtimeclient.ContainerData, error) {
+	containerID, err := runtimeclient.ParseContainerID(runtimeclient.ContainerdName, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, _, _, err := c.getContainerDataAndContainerAndTask(containerID)
+	if err != nil {
+		return nil, err
+	}
+	return containerData, nil
+}
+
+func (c *ContainerdClient) GetContainerDetails(containerID string) (*runtimeclient.ContainerDetailsData, error) {
+	containerID, err := runtimeclient.ParseContainerID(runtimeclient.ContainerdName, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, container, task, err := c.getContainerDataAndContainerAndTask(containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := container.Spec(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("getting spec for container %q: %w", containerID, err)
+	}
+
+	mountData := make([]runtimeclient.ContainerMountData, len(spec.Mounts))
+	for i := range spec.Mounts {
+		mount := spec.Mounts[i]
+		mountData[i] = runtimeclient.ContainerMountData{
+			Source:      mount.Source,
+			Destination: mount.Destination,
+		}
+	}
+
+	return &runtimeclient.ContainerDetailsData{
+		ContainerData: *containerData,
+		Pid:           int(task.Pid()),
```

`task` might be `nil`, and in case it is, `GetContainerDetails` should return an error `got zero pid` as others `ContainerRuntimeClient` do.